app/assets/javascripts/angular/controllers/mobile-dashboard.js

```diff
@@ -0,0 +1,200 @@
+angular.module('mobileApp', ['timer', 'truncate'])
+  .controller('mobileCtrl', ['$scope', '$http', '$interval', '$window', function($scope, $http, $interval, $window) {
+
+  // Initializations
+  $scope.nextRound = new moment();
+  $scope.news = [];
+
+  // Tab changing
+  $scope.focusedTab = 'news';
+
+  $scope.setTab = function(tab) {
+    $scope.focusedTab = tab;
+  }
+
+  // Some stuff shouldn't change. Eg: My Role, My Country.
+  $scope.myCountry = "Germany";
+  $scope.myRole = "Head of State";
+
+
+
+  // $scope.roleColours = {
+  //   'Head of State':
+  // }
+
+
+
+  // ROLE                       SCREENS                                    COLOUR
+// Military                   News, Comms, Operatives, Spying            Red
+// Scientist                  News, Comms, Research, Trades, Rumours     Blue
+// Head of State              News, Comms, Income                        Green
+// Deputy Head of State       News, Comms, Spying                        Dark Grey
+// Ambassador                 News, Comms                                Purple
+// Alien                      News, Comms, Operatives                    Black
+
+
+
+
+  // Mock data for Messages to show the intended structure.
+  $scope.myMessages = [
```

Comment:
Here's the altered data structure for Messages that I have in mind. The messages themselves haven't changed, but what I've done is grouped them together into "conversations". All that it actually adds is a property for a) who the conversation is actually with, and b) when the latest message was sent. But the grouping will be necessary to make this entire thing work, and those two properties just make some things easier on my end.